### DISCUSSION TITLE
Permissions and viewing the Poly lookup

### DISCUSSION THREAD

**Discussion by MB-Motive** [khoait/DCE.PCF#72] [General] [answered]
Hi Khoa

You have made a really great and useful PCF with the PolyLookup (repo starred!)

We have been testing it out for a app we are building. Everything was looking good until we applied security role restrictions (e.g. can only edit records owned by you) - this made the PolyLookup blank and users with read only access can't see the selected items. It also made the outputted text blank (I thought I might be able to show the outputted text if the form was read only).

I saw that Permissions are listed in your To Do. Is the above behaviour expected? Are there workarounds you recommend? Do you know when you are planning to add support for permissions? 

Thanks
Matt

**Comment by khoait** [comment]
Hi Matt, thanks for your interest in the control!

The blank control shouldn't happen if users have read permission on both sides of the association. I'll check again when I find some time this week.

I'm planning to have a new version of PolyLookup supporting connection relationship type this month. Then I can work on handling permissions next month.

In the meantime, you can have a workaround below:
- 2 fields of the same column on the form:
  - 1 field displayed as a normal text field (with comma delimited values)
  - 1 field displayed as PolyLookup
- A form script to check which field to show/hide based on what permissions/roles the user has.

I'll get back to you soon with my updates.

**Comment by khoait** [comment, marked as answer]
Hey Matt, I found it. The control only loads selected items in the Update form.

I'll open a bug for this. Should be a quick fix. I'll make a release for this soon and let you know.

**Reply by khoait** [reply]
bug tracked in #73

**Reply by MB-Motive** [reply]
Hi Khoa. 
Thanks for looking into it so quickly and finding the issue/bug. Look forward to the updated release.
Thanks
Matt

**Reply by khoait** [reply]
Hi Matt. This is fixed in Release v1.0.3.1: https://github.com/khoait/DCE.PCF/releases/tag/v1.0.3.1. Hope it helps!

**Reply by MB-Motive** [reply]
Hi Khoa. Thanks for updating this so quickly. Tested in our environment and works perfectly now for read only records.
